tensorflow/compiler/tf2xla/kernels/image_resize_ops.h

```diff
@@ -66,6 +66,17 @@ class ResizeBilinearGradOp : public XlaOpKernel {
 #endif
 };
 
+class ResizeNearestNeighborGradOp : public XlaOpKernel {
```

Comment:
Do you know if making the TPU kernel just support GPU would work in your case here instead of making a separate GPU one?

https://github.com/tensorflow/tensorflow/blob/41d4cdb273d2ed0e15895e0efe530205b4da699a/tensorflow/core/tpu/kernels/image_resize_ops.cc#L158

Reply:
I don't know, I wasn't aware that this would be a possibility. I assumed it's not possible since that's not what was done in the bilinear case.

Reply:
I tried removing the restriction and added `//tensorflow/core/tpu/kernels:image_resize_ops` to the build targets so it gets loaded. Unfortunately the test seems to fail due to:
```
Traceback (most recent call last):
  File "/home/lgeiger/.cache/bazel/_bazel_lgeiger/8da1bfbfd9537ea4f19db64e1f55bee4/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/compiler/tests/image_ops_jit_compile_test_cpu.runfiles/absl_py/absl/testing/parameterized.py", line 316, in bound_param_test
    return test_method(self, *testcase_params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lgeiger/.cache/bazel/_bazel_lgeiger/8da1bfbfd9537ea4f19db64e1f55bee4/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/compiler/tests/image_ops_jit_compile_test_cpu.runfiles/org_tensorflow/tensorflow/compiler/tests/image_ops_jit_compile_test.py", line 63, in testGradImageResize
    self.assertAllClose(train(x, y), compiled_train(x, y))
                                     ^^^^^^^^^^^^^^^^^^^^
  File "/home/lgeiger/.cache/bazel/_bazel_lgeiger/8da1bfbfd9537ea4f19db64e1f55bee4/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/compiler/tests/image_ops_jit_compile_test_cpu.runfiles/org_tensorflow/tensorflow/python/util/traceback_utils.py", line 141, in error_handler
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/lgeiger/.cache/bazel/_bazel_lgeiger/8da1bfbfd9537ea4f19db64e1f55bee4/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/compiler/tests/image_ops_jit_compile_test_cpu.runfiles/org_tensorflow/tensorflow/python/eager/polymorphic_function/polymorphic_function.py", line 832, in __call__
    result = self._call(*args, **kwds)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lgeiger/.cache/bazel/_bazel_lgeiger/8da1bfbfd9537ea4f19db64e1f55bee4/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/compiler/tests/image_ops_jit_compile_test_cpu.runfiles/org_tensorflow/tensorflow/python/eager/polymorphic_function/polymorphic_function.py", line 918, in _call
    return self._concrete_variable_creation_fn._call_flat(  # pylint: disable=protected-access
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lgeiger/.cache/bazel/_bazel_lgeiger/8da1bfbfd9537ea4f19db64e1f55bee4/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/compiler/tests/image_ops_jit_compile_test_cpu.runfiles/org_tensorflow/tensorflow/python/eager/polymorphic_function/concrete_function.py", line 1323, in _call_flat
    return self._inference_function.call_preflattened(args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lgeiger/.cache/bazel/_bazel_lgeiger/8da1bfbfd9537ea4f19db64e1f55bee4/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/compiler/tests/image_ops_jit_compile_test_cpu.runfiles/org_tensorflow/tensorflow/python/eager/polymorphic_function/atomic_function.py", line 216, in call_preflattened
    flat_outputs = self.call_flat(*args)
                   ^^^^^^^^^^^^^^^^^^^^^
  File "/home/lgeiger/.cache/bazel/_bazel_lgeiger/8da1bfbfd9537ea4f19db64e1f55bee4/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/compiler/tests/image_ops_jit_compile_test_cpu.runfiles/org_tensorflow/tensorflow/python/eager/polymorphic_function/atomic_function.py", line 251, in call_flat
    outputs = self._bound_context.call_function(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lgeiger/.cache/bazel/_bazel_lgeiger/8da1bfbfd9537ea4f19db64e1f55bee4/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/compiler/tests/image_ops_jit_compile_test_cpu.runfiles/org_tensorflow/tensorflow/python/eager/context.py", line 1486, in call_function
    outputs = execute.execute(
              ^^^^^^^^^^^^^^^^
  File "/home/lgeiger/.cache/bazel/_bazel_lgeiger/8da1bfbfd9537ea4f19db64e1f55bee4/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/compiler/tests/image_ops_jit_compile_test_cpu.runfiles/org_tensorflow/tensorflow/python/eager/execute.py", line 53, in quick_execute
    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
tensorflow.python.framework.errors_impl.InvalidArgumentError: Symbol a_inference_train_151__XlaMustCompile_true_config_proto_3175580994766145631_executor_type_11160318154034397263_.112 not found. [Op:__inference_train_151]
```

Any idea where this could come from?

Reply:
Not sure, looping in @Princejain1101  to take a look.

Reply:
Any updates?

Reply:
Ahh sorry for the delay here. Prince was OOO and has just recently left Google. How urgent is this? 

Let me dig around and see why this was originally TPU only. Worst case is to make this implementation GPU only.

Reply:
Ahh I see, so this looks like a case where TPU has a special implementation inside XLA that can't be reused for GPU, so the approach you have here is roughly the right one.